scheduler/src/cook/mesos/constraints.clj

```diff
@@ -116,7 +117,37 @@
   (let [needs-gpus? (job-needs-gpus? job)]
     (->gpu-host-constraint job needs-gpus?)))
 
-(def job-constraint-constructors [build-novel-host-constraint build-gpu-host-constraint])
+(defrecord user-defined-constraint [constraints]
+  JobConstraint
+  (job-constraint-name [this] (get-class-name this))
+  (job-constraint-evaluate
```

Comment:
There is an overloaded version: `[this target-vm-resources target-vm-attrs target-vm-tasks-assigned]` which we are not implementing here. I'm not sure where the overloaded version is implemented?